mm2src/coins/eth/fee_estimation/eth_fee_events.rs

```diff
@@ -0,0 +1,180 @@
+use crate::eth::EthCoin;
+use common::executor::Timer;
+use mm2_event_stream::{Broadcaster, Event, EventStreamer, NoDataIn, StreamHandlerInput};
+
+use async_trait::async_trait;
+use futures::channel::oneshot;
+use instant::Instant;
+use serde::Deserialize;
+use serde_json::Value as Json;
+use std::convert::TryFrom;
+
+#[derive(Deserialize)]
+#[serde(deny_unknown_fields)]
+/// Types of estimators available.
+/// Simple - simple internal gas price estimator based on historical data.
+/// Provider - gas price estimator using external provider (using gas api).
+enum EstimatorType {
+    Simple,
+    Provider,
+}
+
+#[derive(Deserialize)]
+#[serde(deny_unknown_fields, default)]
+struct EthFeeStreamingConfig {
+    /// The time in seconds to wait before re-estimating the gas fees.
+    pub estimate_every: f64,
+    /// The type of the estimator to use.
+    pub estimator_type: EstimatorType,
+}
+
+impl Default for EthFeeStreamingConfig {
+    fn default() -> Self {
+        Self {
+            estimate_every: 15.0,
+            estimator_type: EstimatorType::Simple,
+        }
+    }
+}
+
+pub struct EthFeeEventStreamer {
+    config: EthFeeStreamingConfig,
+    coin: EthCoin,
+}
+
+impl EthFeeEventStreamer {
+    pub fn try_new(config: Option<Json>, coin: EthCoin) -> serde_json::Result<Self> {
+        let config = config.map(serde_json::from_value).unwrap_or(Ok(Default::default()))?;
+
+        Ok(Self { config, coin })
+    }
+}
```

Comment:
no need to return error here. Let's fallback to default as I have highlighted earlier.

Reply:
covered in https://github.com/KomodoPlatform/komodo-defi-framework/pull/2172/commits/4ca4d693783fc2d28a1fb96b553800deba8a097c, by converting try_new to new.